webapp/crashstats/crashstats/utils.py

```diff
@@ -811,45 +823,37 @@ def pci_ids__parse_graphics_devices_iterable(iterable, debug=False):
         #     device  device_name                       <-- single tab indent
         #         subvendor subdevice  subsystem_name   <-- two tab indent
 
+    Since we are only interested in device and vendor ids, we can skip subsystem id lines.
+
     """
     vendor_hex = vendor_name = None
 
     for line in iterable:
         line = smart_str(line)
 
-        if not line:
-            continue
-
-        if line.startswith("#"):
+        if not line or line.startswith("#"):
             if "List of known device classes" in line:
                 # There's a section at the bottom of the files which
                 # we don't need to parse.
                 break
             continue
 
-        match = VENDOR_RE.match(line)
-        if match:
-            vendor_hex, vendor_name = match.group(1, 2)
-            continue
-
-        device_hex = device_name = None
-
-        match = DEVICE_RE.match(line)
-        if match:
-            device_hex, device_name = match.group(1, 2)
-        else:
-            match = SUBDEVICE_RE.match(line)
-            if match:
-                device_hex, device_name = match.group(2, 3)
```

Comment:
This is the part that was wrong -- this is overwriting the device ID and name with the subdevice ID and name.